src/tools/crawler/network.rs

```diff
@@ -16,11 +16,16 @@ pub struct KnownNetwork {
 }
 
 impl KnownNetwork {
-    pub(super) async fn insert_node(&self, addr: SocketAddr) {
+    /// Inserts addr to known_nodes if not yet present (so to avoid overriding the node's statistics).
+    /// Returns true if it's a new node, false otherwise.
+    pub(super) async fn insert_node(&self, addr: SocketAddr) -> bool {
```

Comment:
for later: haven't dug deeper, but perhaps we could use the `Entry` API (with `or_default`) to avoid a double query here

Reply:
I will check this later, thanks for the idea.